pkg/virt-operator/resource/apply/rbac.go

```diff
@@ -206,6 +217,26 @@ func getRoleMetaObject(role interface{}, roleType RoleType) (meta *metav1.Object
 	return
 }
 
+func areRoleRulesEqual(role1 interface{}, role2 interface{}, roleType RoleType) (equal bool) {
+	switch roleType {
+	case TypeRole:
+		role1Obj := role1.(*rbacv1.Role)
+		role2Obj := role2.(*rbacv1.Role)
+		equal = equality.Semantic.DeepEqual(role1Obj.Rules, role2Obj.Rules)
```

Comment:
nit: instead of doing a deep equal here I'd create an ad-hoc function to compare Rules (it's a struct of strings) as using reflection can be 10x slower than a simple comparison (https://rytisbiel.com/2021/03/06/darker-corners-of-go/#deepequal).

Reply:
Great idea! thanks!